runsc/cmd/pidfile.go

```diff
@@ -0,0 +1,58 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+// WritePidFile writes pid file atomically if possible.
+func WritePidFile(path string, pid int) error {
+	pidStr := []byte(strconv.Itoa(pid))
+
+	_, err := os.Stat(path)
+	if err != nil && !os.IsNotExist(err) {
+		// If path exists, write in place, because file could be pipe or something.
+		if err := os.WriteFile(path, pidStr, 0644); err != nil {
+			return fmt.Errorf("failed to write pid file %s: %w", path, err)
+		}
+		return nil
+	} else if err != nil {
```

Comment:
The block ends with `return` now, so replace `} else if err != nil {` with

```
}
if err != nil {
```

Reply:
OK.